concrete/controllers/search/users.php

```diff
@@ -1,42 +1,53 @@
 <?php
+
 namespace Concrete\Controller\Search;
 
 use Concrete\Core\Search\Field\Field\KeywordsField;
-use Concrete\Core\User\Group\GroupList;
+use Doctrine\ORM\EntityManagerInterface;
 
 class Users extends Standard
 {
-
+    /**
+     * @return \Concrete\Controller\Dialog\Search\AdvancedSearch
+     */
     protected function getAdvancedSearchDialogController()
     {
-        return $this->app->make('\Concrete\Controller\Dialog\User\AdvancedSearch');
+        return $this->app->make(\Concrete\Controller\Dialog\User\AdvancedSearch::class);
```

Comment:
I understand using `::class` when you're dealing with the shorthand class name – but this kind of thing drives me up the wall. It's multiple characters _longer_ than the original class. What is the purpose of this?

If you want to do this, you should do it by adding use \Concrete\Controller\Dialog\User\AdvancedSearch above the class, and reference AdvancedSearch::class instead of the fully qualified class name.

Reply:
Speaking for myself, I do prefer the `::class` notation because my IDE recognizes it as a class name (so clicking it leads me to the class definition); this does not happen with strings.

Reply:
Another advantage I think is that if you do automatic refactoring, classes names / FQNs will be updated if needed.

Reply:
It's true that we definitely do not want to be using fully qualified class names. We should always `use` every class so that our dependencies are clearly listed at the top of every file.